WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/ProductPriceSettingsViewController.swift

```diff
@@ -333,17 +333,57 @@ private extension ProductPriceSettingsViewController {
     func configurePrice(cell: UnitInputTableViewCell) {
         let cellViewModel = Product.createRegularPriceViewModel(regularPrice: viewModel.regularPrice, using: CurrencySettings.shared) { [weak self] value in
             self?.viewModel.handleRegularPriceChange(value)
+            self?.setCellPriceAccessibility(price: cell.value, cell: cell)
         }
         cell.selectionStyle = .none
         cell.configure(viewModel: cellViewModel)
+        setCellPriceAccessibility(price: cell.value, cell: cell)
+    }
+
+    func setCellPriceAccessibility(price: String?, cell: UnitInputTableViewCell) {
+        let value = cell.value.isEmpty == false ? cell.value : NSLocalizedString("empty", comment: "Accessibility text for when a price is empty")
+        let currency = cell.value.isEmpty == false ? CurrencySettings.shared.currencyCode.rawValue : ""
+        let accessibilityText = String.localizedStringWithFormat(
+            NSLocalizedString(
+                "Price %1$@ %2$@",
+                comment: "Accessibility label for the regular price cell in Product Price. It's read like: Price 100 dollars."
+            ),
+            value,
+            currency
+        )
```

Comment:
I love how the currency being spoken works here. Good job! 👏 

There is no pause when “Price” and the value is spoken. Typically you would want a pause in between. You can do this by adding a period so it's like:

```
NSLocalizedString("Price. %1$@ %2$@")
```

The best practice IMO though is to use [`accessibilityValue`](https://developer.apple.com/documentation/uikit/uiaccessibilityelement/1619583-accessibilityvalue) and keep the `accessibilityLabel` as static. [Here](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/docs/voiceover-guidelines.md#use-static-labels-for-toggle-buttons) is a related example using toggle buttons. So, the code would be something like this:

```swift
        // static label
        let accessibilityText = NSLocalizedString(
            "Price",
            comment: "Accessibility label for the regular price cell in Product Price. It's read like: Price 100 dollars."
        )

       // The value of the text field here!
       // This is a quick simplified example. Needs refactoring. :D 
        cell.accessibilityValue = String.localizedStringWithFormat(
            NSLocalizedString(
                "%1$@ %2$@",
                comment: "Accessibility label for the regular price cell in Product Price. It's read like: Price 100 dollars."
            ),
            value,
            currency
        )
        cell.setAccessitibily(label: accessibilityText, hint: accessibilityHint)
```

Reply:
TIL'ed about the `accessibilityValue` and the reference is helpful 👍